src/index.ts

```diff
@@ -28,147 +28,91 @@ export interface INetPrinter {
   port: number;
 }
 
-export const USBPrinter = {
-  init: (): Promise<void> =>
-    new Promise((resolve, reject) =>
-      RNUSBPrinter.init(
-        () => resolve(),
-        (error: Error) => reject(error)
-      )
-    ),
-
-  getDeviceList: (): Promise<IUSBPrinter[]> =>
-    new Promise((resolve, reject) =>
-      RNUSBPrinter.getDeviceList(
-        (printers: IUSBPrinter[]) => resolve(printers),
-        (error: Error) => reject(error)
-      )
-    ),
+// Timeout for returning response to client
+const SDK_RESPONSE_TIMEOUT = 5000;
+
+// Promise with timeout
+const promiseWithTimeout = (
+  promise: Promise<INetPrinter>
+): {
+  promiseOrTimeout: Promise<INetPrinter>;
+  timeoutId: ReturnType<typeof setTimeout>;
+} => {
+  let timeoutId;
+  const timeoutPromise: Promise<INetPrinter> = new Promise((_, reject) => {
+    timeoutId = setTimeout(() => {
+      reject(new Error('Request timed out'));
+    }, SDK_RESPONSE_TIMEOUT);
+  });
+  return {
+    promiseOrTimeout: Promise.race([promise, timeoutPromise]),
+    // @ts-ignore
+    timeoutId,
+  };
+};
 
-  connectPrinter: (vendorId: string, productId: string): Promise<IUSBPrinter> =>
-    new Promise((resolve, reject) =>
-      RNUSBPrinter.connectPrinter(
+export const USBPrinter = {
+  connectAndSend: (
+    vendorId: string,
+    productId: number,
+    data: Buffer
+  ): Promise<INetPrinter> => {
+    return new Promise((resolve, reject) =>
+      RNUSBPrinter.connectAndSend(
         vendorId,
         productId,
-        (printer: IUSBPrinter) => resolve(printer),
+        data.toString('base64'),
+        (printer: INetPrinter) => resolve(printer),
         (error: Error) => reject(error)
       )
-    ),
-
-  closeConn: (): Promise<void> =>
-    new Promise((resolve) => {
-      RNUSBPrinter.closeConn();
-      resolve();
-    }),
-
-  print: (text: string): void =>
-    RNUSBPrinter.printRawData(text, (error: Error) => console.warn(error)),
+    );
+  },
 };
 
 export const BLEPrinter = {
-  init: (): Promise<void> =>
-    new Promise((resolve, reject) =>
-      RNBLEPrinter.init(
-        () => resolve(),
-        (error: Error) => reject(error)
-      )
-    ),
-
-  getDeviceList: (): Promise<IBLEPrinter[]> =>
-    new Promise((resolve, reject) =>
-      RNBLEPrinter.getDeviceList(
-        (printers: IBLEPrinter[]) => resolve(printers),
-        (error: Error) => reject(error)
-      )
-    ),
-
-  connectPrinter: (inner_mac_address: string): Promise<IBLEPrinter> =>
-    new Promise((resolve, reject) =>
-      RNBLEPrinter.connectPrinter(
-        inner_mac_address,
-        (printer: IBLEPrinter) => resolve(printer),
-        (error: Error) => reject(error)
+  connectAndSend: (bdAddress: string, data: Buffer): Promise<INetPrinter> => {
+    const { promiseOrTimeout, timeoutId } = promiseWithTimeout(
```

Comment:
i have this question: dont we need method for ble device discovery!

Reply:
We will add that in the future.

For now, we add the bdAddress directly